### PR TITLE
Problem: redundant copy in check validate

### PR DESCRIPTION
**Pull request by mmsqe**
* check validity with extractSliceBytes to avoid copying slice data.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved key comparison accuracy in the database iterator to enhance overall reliability.
	- Enhanced efficiency by reducing unnecessary data copying during slice operations in the iterator.

- **Documentation**
	- Updated CHANGELOG to reflect recent improvements in slice data handling and efficiency enhancements.

### CODE REVIEW

**Review by yihuang — CHANGES_REQUESTED** [review]
we still need to call `Free` somewhere.

**Review by yihuang — APPROVED** [review]

**Comment by mmsqe** [comment]
seems check Exists before get data?

File: versiondb/tsrocksdb/iterator.go [hunk 1]
```diff
@@ -23,8 +23,9 @@ func newRocksDBIterator(source *grocksdb.Iterator, prefix, start, end []byte, is
 		} else {
 			source.Seek(end)
 			if source.Valid() {
-				eoakey := moveSliceToBytes(source.Key()) // end or after key
-				if bytes.Compare(end, eoakey) <= 0 {
+				eoakey := source.Key() // end or after key
+				defer eoakey.Free()
```
Comment (mmsqe) [inline versiondb/tsrocksdb/iterator.go:27]: not sure if defer here since source.Prev will change

Comment (yihuang) [reply versiondb/tsrocksdb/iterator.go:27]: the memory is either managed by the caller or not, if it's not managed by the caller, then the caller should not free it, vice versa.

**Pull request merged**